Gordon360/Controllers/RecIM/TeamsController.cs

```diff
@@ -0,0 +1,282 @@
+        /// <returns></returns>
+        [HttpPost]
+        [Route("")]
+        public async Task<ActionResult<TeamViewModel>> CreateTeam([FromQuery] string username, TeamUploadViewModel newTeam)
+        {
+            var activity = _activityService.GetActivityByID(newTeam.ActivityID);
+            if (activity is null)
+                return UnprocessableEntity($"This activity does not exist");
+            if (_activityService.ActivityTeamCapacityReached(newTeam.ActivityID))
+                return UnprocessableEntity($"The activity has reached the maximum team capacity");
+            if (_teamService.HasTeamNameTaken(newTeam.ActivityID, newTeam.Name))
+                return UnprocessableEntity($"Team name {newTeam.Name} has already been taken by another team in this activity");
+           //redudant check for API as countermeasure against postman navigation around UI check, admins can make any number of teams
+            if (_teamService.HasUserJoined(newTeam.ActivityID, username) && !_participantService.IsAdmin(username))
+                return UnprocessableEntity($"Participant {username} already is a part of a team in this activity");
+            if(_activityService.ActivityRegistrationClosed(newTeam.ActivityID) && !_participantService.IsAdmin(username))
+                return UnprocessableEntity("Activity Registration has closed.");
+            if (_activityService.ActivityTeamCapacityReached(newTeam.ActivityID))
+                return UnprocessableEntity("Activity capacity has been reached. Try again later.");
+
+            try
+            {
+                var team = await _teamService.PostTeamAsync(newTeam, username);
+                // future error handling
+                // (cannot implement at the moment as we only have 4 developer accs)
+                if (team is null)
+                {
+                    return BadRequest($"Participant {username} already is a part of a team in this activity");
+                }
+                return CreatedAtAction("CreateTeam", team);
+            }
+            catch (Exception)
+            {
+                throw;
+            }
+            
+        }
+
+        /// <summary>
+        /// Add a participant to a team
+        /// </summary>
+        /// <param name="teamID"></param>
+        /// <param name="participant">Default Role Value value 3 (Member)</param>
+        /// <returns></returns>
+        [HttpPost]
+        [Route("{teamID}/participants")]
+        [StateYourBusiness(operation = Operation.UPDATE, resource = Resource.RECIM_TEAM)]
+        public async Task<ActionResult<ParticipantTeamViewModel>> AddParticipantToTeam(int teamID, ParticipantTeamUploadViewModel participant)
+        {
+            var inviterUsername = AuthUtils.GetUsername(User);
+            var activityID = _teamService.GetTeamActivityID(teamID);
+            if (!_teamService.HasUserJoined(activityID, participant.Username) || _participantService.IsAdmin(inviterUsername))
```

Comment:
Never really thought about it like that, I just mentally thought that admins should be able to have full free reign over overwriting common rules. But I suppose in this situation, this would still be incredibly awkward.

Reply:
I suppose there _could_ be a little rework here. The main usecase of this is actually so that admins can add the same person to **multiple** teams in the same activity. I updated TeamService to return null on AddParticipantToTeamAsync if the participant is in the team.

Reply:
I think maybe something was miscommunicated, I asked:
> Do you really want admins to be able to add users to multiple teams in the same activity?

and you responded:
> The main usecase of this is actually so that admins can add the same person to multiple teams in the same activity.

So, the answer to my question is **yes**, you do want admins to be able to add users to multiple teams in the same activity?